homeassistant/components/suez_water/config_flow.py

```diff
@@ -37,16 +37,19 @@ async def validate_input(data: dict[str, Any]) -> None:
             data[CONF_PASSWORD],
             counter_id,
         )
-        if not await client.check_credentials():
-            raise InvalidAuth
-    except PySuezError as ex:
-        raise CannotConnect from ex
-
-    if counter_id is None:
         try:
-            data[CONF_COUNTER_ID] = await client.find_counter()
+            if not await client.check_credentials():
+                raise InvalidAuth
         except PySuezError as ex:
-            raise CounterNotFound from ex
+            raise CannotConnect from ex
+
+        if counter_id is None:
```

Comment:
I don't think the changes here are related to any of the two mentioned topics from the PR description

Reply:
Yep you're right, that's a bugfix I forgot to split.
I'll draft the PR until I have time to do it

Reply:
I would even go so far as to suggest to split this PR and make one for runtime_data, and one for data description.
I would have a few comments for runtime_data...

Reply:
Ouch, I thought that I understood what I had done.
But no problem to give it it's own PR, even tough I'm a little bit worried 😅

Reply:
Only data_descriptions remains here